### DISCUSSION TITLE
Can I use a post request to update a tables data?

### DISCUSSION THREAD

**Discussion by SnappierSoap318** [supabase/supabase#7686] [Questions]
I'm using a SIM Module with an Arduino to update a supabase table, but unfortunately it can't do Patch requests, is it possible to update a table using post request?

**Comment by NanoCode012** [comment]
As I understand, `rpc` uses POST, perhaps you can create a database function (that updates the table) and call it from your device?

**Reply by SnappierSoap318** [reply]
so i would need to start a client inside the function right? or is there a way to access tables in functions

**Reply by NanoCode012** [reply]
Nope, "Database function" is just pure SQL that runs inside your database. This is not to be confused with "Edge functions" which is a different thing (which also works for you and requires to connect to a client).

According to doc (https://supabase.com/docs/guides/functions#database-functions-vs-edge-functions), 
- data intensive -> database functions
- low latency -> edge functions

From your requirements of sending from an Arduino, may I suggest Edge functions instead of my earlier Database functions suggestion?

https://supabase.com/docs/guides/functions

**Reply by SnappierSoap318** [reply]
yes I am using Edge function because of the low latency. 

But i am getting this while trying to setup functions 

`error: TS2552 [ERROR]: Cannot find name SUPABASE_URL.`

![image](https://user-images.githubusercontent.com/42905211/179037172-29c72b52-5a8b-44ef-9e5a-0b2769574e3c.png)

**Reply by NanoCode012** [reply]
I believe it should be `"SUPABASE_URL"` with quotes and not `SUPABASE_URL`.

See example: https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/_shared/supabaseClient.ts

**Reply by SnappierSoap318** [reply]
![image](https://user-images.githubusercontent.com/42905211/179047723-408b1740-6c69-43a2-840d-6866780d3481.png)
Getting this error even though Auth header is present

**Reply by SnappierSoap318** [reply]
![image](https://user-images.githubusercontent.com/42905211/179051565-ce3e5537-809e-48f2-9cd8-f5cd410878fa.png)
ok im getting this now
